src/domain/request/usecase/ExecuteReadyRequestsUseCase.js

```diff
@@ -0,0 +1,45 @@
+class ExecuteReadyRequestsUseCase {
+  constructor(
+    fetchDataUseCase,
+    selectDataUseCase,
+    sendResponseToOracleUseCase,
+    requestRepository,
+    responseRepository,
+    logger,
+  ) {
+    this.fetchDataUseCase = fetchDataUseCase;
+    this.selectDataUseCase = selectDataUseCase;
+    this.sendResponseToOracleUseCase = sendResponseToOracleUseCase;
+    this.requestRepository = requestRepository;
+    this.responseRepository = responseRepository;
+    this.logger = logger;
+  }
+
+  async executeReadyRequests() {
+    const requests = await this.requestRepository.getReadyRequests();
+
+    const promises = requests.map(async (request) => {
+      request.state.markAsProcessed();
```

Comment:
Ephemeral state. Not saved in the database. Other worker can pick up this request and work on it too.